design-toolkit/components/src/components/dialog/index.tsx

```diff
@@ -0,0 +1,247 @@
+}
+
+const DialogBody = forwardRef<HTMLDivElement, DialogBodyProps>(
+  ({ children, ...rest }, ref) => {
+    const {
+      size,
+      isDismissable,
+      isOpen,
+      onOpenChange,
+      parentRef,
+      isKeyboardDismissDisabled,
+    } = useDialogContext();
+    const isSSR = useIsSSR();
+    const [portal, setPortal] = useState(isSSR ? null : document.body);
+
+    useEffect(() => {
+      const node = parentRef?.current;
+      const port = isSSR ? null : document.createElement('div');
+
+      if (node && port) {
+        node.appendChild(port);
+        setPortal(port);
+      }
+
+      return () => {
+        port?.remove();
+
+        setPortal(isSSR ? null : document.body);
+      };
+    }, [isSSR, parentRef]);
+
+    return portal ? (
+      <RACModalOverlay
+        UNSTABLE_portalContainer={portal}
+        isKeyboardDismissDisabled={!isDismissable && isKeyboardDismissDisabled}
+        isOpen={isOpen}
+        onOpenChange={onOpenChange}
+        isDismissable={isDismissable}
+        className='absolute inset-0 flex items-center justify-center'
+        {...rest}
+      >
+        <RACModal
+          ref={ref}
+          className={cn(
+            'flex flex-col justify-center bg-surface-overlay align-start',
+            size === 'sm' && 'rounded-medium',
+            size === 'lg' && 'rounded-large',
+          )}
+        >
+          <RACDialog className={dialogClasses({ size })}>{children}</RACDialog>
+        </RACModal>
+      </RACModalOverlay>
+    ) : null;
+  },
+);
+DialogBody.displayName = 'DialogBody';
+
+const DialogContent = ({
+  children,
+  className,
+}: PropsWithChildren<{ className?: string }>) => {
+  const { size } = useDialogContext();
+  return (
+    <div
+      className={cn(
+        'flex flex-col',
+        size === 'sm' && 'gap-xs',
+        size === 'lg' && 'gap-l',
+        className,
+      )}
+    >
+      {children}
+    </div>
+  );
+};
+
+export const DialogTitle = ({
+  children,
+  className,
+}: PropsWithChildren<{ className?: string }>) => {
+  const { size } = useDialogContext();
+  return (
+    <RACHeading
+      slot='title'
+      className={cn(
+        'text-default-light',
+        size === 'sm' && 'mb-s text-header-m',
+        size === 'lg' && 'mb-m text-header-l',
+        className,
+      )}
+    >
+      {children}
+    </RACHeading>
+  );
+};
+
+const DialogButton = ({ children, ...props }: ButtonProps) => {
+  const { size: dialogSize } = useDialogContext();
+  const buttonSize = buttonSizes[dialogSize];
+  return (
+    <Button size={buttonSize} {...props}>
+      {children}
+    </Button>
+  );
+};
+
+const DialogFooter = ({
+  children,
+  className,
+}: PropsWithChildren<{ className?: string }>) => {
+  const { size } = useDialogContext();
+  return (
+    <div
+      className={cn(
+        'flex justify-end gap-xs',
+        //margin is hardcoded to match the spacing values
+        size === 'sm' && 'mt-[16px]',
+        size === 'lg' && 'mt-[24px]',
```

Comment:
You can actually use tailwind to do these without hardcoding the values, so they are consistent with the theme:

```suggestion
        size === 'sm' && 'mt-l',
        size === 'lg' && 'mt-xl',
```

Reply:
Good call, I'm not sure how I missed that!